### PR TITLE
Use ISO8601 timestamps for the CalDAV trashbin

### PR DESCRIPTION
**Pull request by ChristophWurst**
Addresses https://github.com/nextcloud/cdav-library/pull/474#discussion_r641543953

Ref https://github.com/nextcloud/cdav-library/pull/503

### CODE REVIEW

File: apps/dav/lib/CalDAV/Calendar.php [hunk 1]
```diff
@@ -63,6 +65,11 @@ class Calendar extends \Sabre\CalDAV\Calendar implements IRestorable, IShareable
 	 * @param IConfig $config
 	 */
 	public function __construct(BackendInterface $caldavBackend, $calendarInfo, IL10N $l10n, IConfig $config) {
+		// Convert deletion date to ISO8601 string
+		if (isset($calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT])) {
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Calendar.php:69]: Maybe add an alias to shorten all these calls.

File: apps/dav/lib/CalDAV/Trashbin/Plugin.php [hunk 2]
```diff
@@ -101,7 +103,12 @@ private function propFind(
 		INode $node): void {
 		if ($node instanceof DeletedCalendarObject) {
 			$propFind->handle(self::PROPERTY_DELETED_AT, function () use ($node) {
-				return $node->getDeletedAt();
+				$ts = $node->getDeletedAt();
+				if ($ts === null) {
+					return null;
+				}
+
+				return (new DateTimeImmutable('@' . $ts))->format(DateTimeInterface::ATOM);
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Trashbin/Plugin.php:111]:
```suggestion
				return (new \DateTimeImmutable())->setTimestamp($ts)->format(DateTimeInterface::ATOM);
```
Same

File: apps/dav/lib/CalDAV/Calendar.php [hunk 3]
```diff
@@ -63,6 +65,11 @@ class Calendar extends \Sabre\CalDAV\Calendar implements IRestorable, IShareable
...
+			$calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT] = (new DateTimeImmutable('@' . $calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT]))->format(DateTimeInterface::ATOM);
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Calendar.php:70]:
```suggestion
			$calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT] = (new DateTimeImmutable())->setTimestamp($calendarInfo[\OCA\DAV\CalDAV\Trashbin\Plugin::PROPERTY_DELETED_AT]))->format(DateTimeInterface::ATOM);
```
Maybe cleaner. No big deal.

**Review by tcitworld — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]
Code makes sense :+1:

**Pull request merged**